smart_open/tests/test_smart_open.py

```diff
@@ -462,11 +462,17 @@ def test_s3_iter_bucket_moto(self):
                 expected[key_name] = content
 
         # read all keys + their content back, in parallel, using s3_iter_bucket
-        result = dict(smart_open.s3_iter_bucket(mybucket))
+        result = {}
+        for k, c in smart_open.s3_iter_bucket(mybucket):
+            result[k.name] = c
+        #result = dict(smart_open.s3_iter_bucket(mybucket))
```

Comment:
We don't want to keep dead code around, even as comments. The line is preserved in git history, if we ever need it again.